common/djangoapps/track/backends/oli.py

```diff
@@ -0,0 +1,124 @@
+"""OLI analytics service event tracker backend."""
+
+from __future__ import absolute_import
+
+import json
+import logging
+
+from requests_oauthlib import OAuth1Session
+
+from django.contrib.auth.models import User
+
+from student.models import anonymous_id_for_user
+from track.backends import BaseBackend
+
+
+LOG = logging.getLogger(__name__)
+
+
+class OLIAnalyticsBackend(BaseBackend):
+    """
+    Transmit events to the OLI analytics service
+    """
+    def __init__(self, **kwargs):
+        super(OLIAnalyticsBackend, self).__init__(**kwargs)
+
+        # the oli analytics service endpoint will be at
+        # "{}{}".format(oli_analytics_service_url, path)
+        self.oli_analytics_service_url = kwargs.get('oli_analytics_service_url', '')
+        self.path = kwargs.get('path', '')
```

Comment:
Do these instance variables _need_ to be set to the empty string or
could they simply be left as `None`?

```python
self.path = kwargs.get('path') # default to `None`
```

While both values will evaluate to `Falsey`, allowing us to do checks
like `if self.path:`, I find that each offers a slight difference in
semantics.

When I see a variable set to the emtpy string `''`, that reads as "this
has a value and that value is zero characters long; it implies that a
value is being explicitly set.

When I see a variable set to `None`, that reads as "this does _not_ have
a value assigned to it"; it implies that a value has yet to be
initialized.

In this case, we're (currently) setting the default value to `''` if one
is not passed into the constructor. I would argue that `None` is more
apt here as it's value is "nothing", as opposed to "empty".